### PR TITLE
Basic session param override struct

### PR DESCRIPTION
**Pull request by aforge**
This PR is purposed to establish a general pattern to override session param.

### CODE REVIEW

File: server/datamodel.go [hunk 1]
```diff
@@ -518,6 +518,13 @@ type MsgServerInfo struct {
 	SeqId int `json:"seq,omitempty"`
 }
 
+// Session parameter overrides.
+// Used when by the remote topic masters to impersonate multiple proxied sessions.
```
Comment (or-else) [inline server/datamodel.go:522]: Used ~when~ by

File: server/datamodel.go [hunk 2]
```diff
@@ -518,6 +518,13 @@ type MsgServerInfo struct {
...
+type sessionOverrides struct {
```
Comment (or-else) [inline server/datamodel.go:523]: The alternative would be to use the regular `Session` with `proto == PROXY` or `OVERRIDE` or something like that. `sessOverrides` requires passing it in exactly the same places where `sess` is already passed. It tells me extending `Session` is better than splitting it into two. 

The number of proxied sessions is T * C, where N is the number of topics and C is the number of cluster nodes, exactly the same as the increase in the number of topics due to clustering, which looks acceptable to me. Then we leave most parameters of the overridden session as blank so they won't take up memory.

Comment (aforge) [reply server/datamodel.go:523]: Yes, I considered extending `Session`.
Right now, I'm planning to use one `Session` object per master topic and multiplex multiple remote sessions in this session.
So if two requests to the same topic arrive from two different sessions, they will have to share their shared proxy session on the master. So modifying the state of the shared session isn't going to work here. Or did you mean something else?

Another thing I was thinking about is a thread local variable/context. `Topic.run()` is single-threaded. We can set the context when an event is polled out of the queue and clear it after the event has been processed.

File: server/datamodel.go (hunk 1 above)
Comment (aforge) [reply server/datamodel.go]: Done.

**Comment by aforge** [comment]
I'm planning to use `sessionOverrides` for more session-level params such as `authLvl` and device settings.
As for `queueOutWithOverrides`, I'll change the existing method in follow-up PRs.

**Pull request merged**